modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/runtime/transaction/TransactionBindingDelegate.java

```diff
@@ -120,7 +120,11 @@ public void invalidate() {
       }
     } finally {
       if (!bound) {
-        connectionHandler.release();
+        try {
+          connectionHandler.release();
+        } catch (Exception e) {
+          // Nothing to do
```

Comment:
I would log a warning in an general exception is enable easier troubleshooting in case something results in a leak.